llvm/tools/objwriter/objwriter.h

```diff
@@ -100,6 +100,7 @@ class ObjectWriter {
   void EmitCFIEnd(int Offset);
   void EmitCFILsda(const char *LsdaBlobSymbolName);
   void EmitCFICode(int Offset, const char *Blob);
+  void EmitCFICompactUnwindEncoding(unsigned int Encoding);
```

Comment:
So, as I understand - the idea is to switch the default to be DWARF, and this will be a way to request compact format, if so desired in the future.

Reply:
Correct

Reply:
Could you please document this in the code?

(However, it is not clear to me yet if I'm asking for a quick comment here about the behavior of (zero -> attempt, non-zero -> provided) or more of your general explanation to be included somewhere else.  I'll continue reading but am certainly open to a suggestion.)

Reply:
Sure. I will try to do that tomorrow.

Reply:
Thanks!  To verify my understanding:

The Streamer is changed from "always try to infer" to "infer if zero".

The objwriter abstraction on top of it changes to a default of the `...EHFrameOnly` value (which is not zero as defined in the header file but the specific `UNWIND_...` value written in `MCObjectFileInfo.cpp`).  That can be overridden to zero (old default - infer) or any non-zero value (just use it).

I assume that any producer has to be aware of the special values such as `UNWIND_...` and use a different range of numbers.